src/Tiqr/Legacy/TiqrService.php

```diff
@@ -428,4 +429,10 @@ private function initSession(): void
     {
         $this->session = $this->requestStack->getSession();
     }
+
+    public function stateStorageHealthCheck(string &$message = ''): bool
+    {
+        assert($this->tiqrStateStorage instanceof  Tiqr_HealthCheck_Interface);
```

Comment:
Do we want to fail the GSSP this rigorously when there is no health check compatible storage backend? Maybe simply continue without the check, or log the situation? 

Please consult with the PM what the desired behaviour is in this situation. And perhaps update the story and/or PR description accordingly.

Reply:
I'm fine with failing here. We expect the statestorage to provide a healthcheck interface. This will make it obvious when it's not there. It's not like the interface is suddenly going to dissappear at runtime, causing an outage.

Reply:
Thanks for your insight @pmeulen